src/extended_json.ts

```diff
@@ -234,6 +234,13 @@ function serializeValue(value: any, options: EJSONSerializeOptions): any {
     return { $numberDouble: Object.is(value, -0) ? '-0.0' : value.toString() };
   }
 
+  if (typeof value === 'bigint') {
+    if (!options.relaxed) {
```

Comment:
so this is pretty last minute in the review process, so this is more of a suggestion / general idea.  if you want to make the refactor, we should run it by the team since it deviates from the existing pattern here and we'd want to make sure everyone is alright with it.

An alternate approach here could be converting the bigint into a long and then stringify-ing the long:

```typescript
return Long.fromBigInt(value).toExtendedJSON();
```

this isn't exactly as straightforward as the code snippet because

1. we'd want to do the truncation in `Long.fromBigInt`, but that's already public api so we can't change the behavior (although, we _could_, just a larger scope)
2. so we may want to create some other internal static factory on the Long class

But the benefit to this approach is that we're guaranteed that the actual stringification is exactly the same for big ints and longs (which is what we want conceptually).

Reply:
https://jira.mongodb.org/browse/NODE-4954
Filed this to improve the way we obtain a long from a bigint. Reusing the current logic would mean stringifying the number to calculate the high/low bits only to stringify it again, and undoing that here and confirming total test coverage takes the scope a bit further than necessary for the goal here

Reply:
Can you clarify?  I don't see why this requires additional test coverage.

This entirely reuses existing Long construction logic (presumably tested) and ejson stringification logic (tested).  The only new behavior we'd need to test is that bigints > 64 bits are truncated properly (straightforward to test).

Reply:
True there's implicit coverage through the EJSON tests, good catch. Still the string -> number -> string steps are unnecessary we can avoid those translation steps. long.toString even references div operations being expensive in code comments

Reply:
I think I agree with Neal here. The existing logic for `Long.fromBigInt` definitely does work, but it is also as Neal said, the code path is overly complicated and repeats work. The code path is `Long.fromBigInt` which has to call `BigInt.toString`, passing that result back to `Long.fromString` and then going ahead and stringifying the resulting `Long`. The time loss from repeating this work might be negligible, but I think in principle it still is preferable to avoid this.